### PR TITLE
Feature sb 66 vendor page

### PR DESCRIPTION
**Pull request by ValentinaTenus**
1. add vendor profile page visible for buyers
2. user can see vendor's rating
3. click on follow and unfollow make reqest follow vendor or unfollow
4. by default active vendor's products are shown
5. when reviews open buyer can see recieved by vendor review with rating. 
JIRA https://team-jfqe9wlknaql.atlassian.net/browse/SB-20
image
image
image

changed buttons image

### CODE REVIEW

File: src/libs/constants/app-route.ts [hunk 1]
```diff
@@ -15,6 +15,8 @@ const AppRoute = {
 	SIZE: "/size",
 	SIZE_GUIDE: "/size-guide",
 	TERMS_OF_USE: "/terms-of-use",
+	VENDOR_PROFILE: "/vendors/:id",
```
Comment (VladManukian) [inline src/libs/constants/app-route.ts]: One of this
```suggestion
	VENDORS_PROFILE: "/vendors/:id",
```
```suggestion
	VENDOR_PROFILE: "/vendor/:id",
```

File: src/pages/vendor-profile/components/vendor-buttons-group/index.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+				variant={isProductShown ? "primary_black_regular" : "secondary_black"}
+			>
+				{t("VendorProfilePage.products")}
+			</StyledButton>
+			<StyledButton
+				onClick={onReviewsClick}
+				// sx={{
+				// 	backgroundColor: colors.lightGray,
+				// 	border: "none",
+				// }}
```
Comment (VladManukian) [inline src/pages/vendor-profile/components/vendor-buttons-group/index.tsx]:
```suggestion

```

**Comment by VladManukian** [comment]
Figma style seems like its tabs, not separate buttons. We need or use gap between buttons, or create a tabs for it.
buttonFigma
buttonYours

File: src/pages/vendor-profile/components/review-item/index.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+	}, [review]);
+
+	const toggleExpanded = useCallback(() => {
+		setIsReviewExpanded(!isReviewExpanded);
+	}, [isReviewExpanded]);
+
+	const shortenReview = useCallback((reviewText: string) => {
+		if (reviewText.length > maxReviewLength) {
+			return reviewText.slice(zeroIndex, maxReviewLength);
+		}
```
Comment (likeri29) [inline src/pages/vendor-profile/components/review-item/index.tsx:45]: Types for return for the two functions above

File: src/pages/vendor-profile/components/header-link-group/index.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+	const { t } = useTranslation();
+
+	return (
+		<StyledHeaderLinksGroup>
+			<StyledHeaderLink href={AppRoute.HOME}>
+				{t("VendorProfilePage.home")}
+			</StyledHeaderLink>
+			<ArrowForwardIosIcon
+				color="secondary"
+				sx={{ fontSize: "12px", height: "100%" }}
```
Comment (VladManukian) [inline src/pages/vendor-profile/components/header-link-group/index.tsx]: Use const for both here

File: src/pages/vendor-profile/components/vendor-buttons-group/index.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
+import { Typography } from "@mui/material";
+import React from "react";
+import { useTranslation } from "react-i18next";
+
+import { StyledButton, StyledVendorsButtonsGroup } from "./styles.ts";
+
+const zIndexZero = 0;
+const zIndexOne = 1;
```
Comment (VladManukian) [inline src/pages/vendor-profile/components/vendor-buttons-group/index.tsx]: Scale theme and create const for this.
Use 0-10 for now, we need decide later, how better.

**Review by VladManukian — APPROVED** [review]

**Review by likeri29 — APPROVED** [review]

**Pull request merged**